packages/admin/dashboard/src/components/search/use-search-results.tsx

```diff
@@ -0,0 +1,750 @@
+import { HttpTypes } from "@medusajs/types"
+import { keepPreviousData } from "@tanstack/react-query"
+import { TFunction } from "i18next"
+import { useCallback, useEffect, useMemo, useState } from "react"
+import { useTranslation } from "react-i18next"
+import {
+  useApiKeys,
+  useCampaigns,
+  useCollections,
+  useCustomerGroups,
+  useCustomers,
+  useInventoryItems,
+  useOrders,
+  usePriceLists,
+  useProductCategories,
+  useProducts,
+  useProductTags,
+  useProductTypes,
+  usePromotions,
+  useRegions,
+  useSalesChannels,
+  useShippingProfiles,
+  useStockLocations,
+  useTaxRegions,
+  useUsers,
+  useVariants,
+} from "../../hooks/api"
+import { useReturnReasons } from "../../hooks/api/return-reasons"
+import { Shortcut, ShortcutType } from "../../providers/keybind-provider"
+import { useGlobalShortcuts } from "../../providers/keybind-provider/hooks"
+import { DynamicSearchResult, SearchArea } from "./types"
+
+type UseSearchProps = {
+  q?: string
+  limit: number
+  area?: SearchArea
+}
+
+export const useSearchResults = ({
```

Comment:
I noticed that sometimes results are shown only while typing i.e. a fetch is running. If I type an email from an order, results are shown only while the loader is visible and when fetching is done "no results" is rendered.